onnxruntime/core/providers/nuphar/scripts/create_shared.sh

```diff
@@ -50,6 +52,18 @@ __EOF__
     rm checksum.cc
 fi
 
+# generate cache_version.cc
+VERSION_FILE="${SCRIPT_DIR}/NUPHAR_CACHE_VERSION"
+cat > $CACHE_DIR/cache_version.cc <<__EOF__
+#include "$VERSION_FILE"
+extern "C"
+const char* _ORTInternal_GetCacheVersion() {
+  return __NUPHAR_CACHE_VERSION__;
+}
+__EOF__
+g++ -std=c++14 -fPIC -o cache_version.o -c cache_version.cc
+rm cache_version.cc
```

Comment:
can you have a single loop to compile all cc files, so it would be easier to add cc files in future?